atomic_reactor/plugin.py

```diff
@@ -354,6 +381,57 @@ def __init__(self, dt, workflow, plugins_conf, *args, **kwargs):
         self.plugins_results = workflow.prebuild_results
         super(PreBuildPluginsRunner, self).__init__(dt, workflow, 'PreBuildPlugin', plugins_conf, *args, **kwargs)
 
+
+class BuildStepPlugin(BuildPlugin):
+    pass
```

Comment:
How about setting the default `is_allow_to_fail` to `False` for build-step plugins, and not setting it in docker_api?